weblate/addons/git.py

```diff
@@ -60,6 +60,50 @@ def get_filenames(self, component):
                 languages[code].extend(translation.filenames)
         return languages
 
+    def get_squash_commit_message(self, repository, log_format, remote, filenames=None):
+        commit_message = self.instance.configuration.get("commit_message")
+
+        if not commit_message:
+            command = [
+                "log",
+                "--format={}".format(log_format),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            commit_message = repository.execute(command)
+
+        if self.instance.configuration.get("append_trailers"):
+            command = [
+                "log",
+                "--format=%(trailers)",
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            trailer_lines = OrderedDict.fromkeys(
```

Comment:
It is not necessary to use OrderedDict, as Weblate requires Python 3.6+ and that has ordered dict, see https://stackoverflow.com/a/39980744/225718 (but it should do no harm).